### ISSUE TITLE
[java] MethodReturnsInternalArray doesn't consider anonymous classes

### ISSUE THREAD

**Issue by adangel** [pmd/pmd#3630]
**Affects PMD Version:** 6.40.0

**Rule:** [MethodReturnsInternalArray](https://pmd.github.io/latest/pmd_rules_java_bestpractices.html#methodreturnsinternalarray)

**Description:**

Methods in anonymous classes are not considered.

**Code Sample demonstrating the issue:**

```java
public class Outer {
    private static final String[] names = new String[] {"a", "b"};
    public static Provider getProvider() {
        return new Provider() {
            @Override
            public String[] getNames() {
                return names; // expected a violation here
            }
        };
    }
    public interface Provider {
        String[] getNames();
    }
}
```

**Expected outcome:**

PMD should report a violation at line 7, but doesn't. This is a false-negative.

Note: already fixed by #3619
